kem/xkem/doc.go

```diff
@@ -0,0 +1,20 @@
+// Package xkem implements KEM based on X25519 and X448.
+package xkem
+
+import (
+	"crypto"
+
+	"github.com/cloudflare/circl/dh/x25519"
+	"github.com/cloudflare/circl/dh/x448"
+	"github.com/cloudflare/circl/kem"
+)
+
+// X25519HkdfSha256 is a KEM using X25519 Diffie-Hellman function with HKDF based on SHA-256.
+func X25519HkdfSha256() kem.AuthScheme {
+	return xkem{x25519.Size, 0x20, "KemX25519HkdfSha256", crypto.SHA256}
```

Comment:
(For KEM TLS) we need that for two different PublicKeys of the same scheme that `pk1.Scheme() == pk2.Scheme()`.  That's a pointer comparison.  If they were created using different invocations of `X25519HkdfSha256()` this won't hold.  I would suggest:

```go
var x25519HkdfSha256 kem.AuthScheme = &xkem{x25519.Size, 0x20, "KemX25519HkdfSha256", crypto.SHA256}
func X25519HkdfSha256() kem.AuthScheme { return x25519HkdfSha256 }
```